src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -163,6 +165,9 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
             _workspaceToLspSolution[workspace] = null;
         }
 
+        // Send a workspace changed notification to anyone subscribed. For example, this is important for semantic tokens refresh.
+        LspWorkspaceChanged?.Invoke(sender, e);
```

Comment:
We may want to document that the event handler implementation for this should not do basically any synchronous work, as that will block here.

Right now the only implementation basically does nothing (good), but I'm wondering if there is a way to ensure this cannot block even if the implementation of the event does something bad.  Async callback function is an obvious option, but doesn't allow for easy registration of listeners (@CyrusNajmabadi for ideas)